src/ae_kqueue.c

```diff
@@ -74,13 +111,17 @@ static int aeApiCreate(aeEventLoop *eventLoop) {
         zfree(state);
         return -1;
     }
+    /* 避免 fd 泄漏, 设置 kqueue fd 运行时可关闭 */
     anetCloexec(state->kqfd);
     state->eventsMask = zmalloc(EVENT_MASK_MALLOC_SIZE(eventLoop->setsize));
+    /* 掩码数组的比特位都初始化为 0 */
     memset(state->eventsMask, 0, EVENT_MASK_MALLOC_SIZE(eventLoop->setsize));
     eventLoop->apidata = state;
     return 0;
 }
 
+/* 按照给定的 setsieze, 重新分配 kqueue event 数组和相对应掩码数组的大小
+ * Question: 为什么需要将数组 state->eventsMask 重新设置为 0 */
```

Comment:
> Question: 为什么需要将数组 state->eventsMask 重新设置为 0

这边我猜，本身 resize 的话，就需要说对这些掩码重新初始化置0，无论是往大的 resize 还是往小的 resize，然后这个置 0 其实应该也不咋影响，因为在 aeApiPoll 里每次对事件处理的时候，都会重新计算每个事件的 mask 然后进行赋值

```suggestion
/* 按照给定的 setsieze，重新分配 kqueue event 数组和相对应掩码数组的大小 */
```

Reply:
看了下代码，确实是每次 eventLoop 都会重新取事件并计算 mask 的值，谢谢解答